engine/api/workflow_queue.go

```diff
@@ -349,7 +349,8 @@ func (api *API) postWorkflowJobResultHandler() Handler {
 		if err := UnmarshalBody(r, &res); err != nil {
 			return sdk.WrapError(err, "postWorkflowJobResultHandler> cannot unmarshal request")
 		}
-		dbWithCtx := api.mustDBWithCtx(ctx)
+		customCtx, _ := context.WithTimeout(context.Background(), 90*time.Second)
```

Comment:
defer call the cancel func, to avoid having a timer pending for a while

Reply:
Nope I can't because it will be almost the same than create a db connection with request context

Reply:
all of this is so sad

Reply:
need to be fixed as discussed